joomla/joomla-cms
# Reinstate electric calendar functionality in date picker (fixes BC break)

Pull Request for Issue # .

### Summary of Changes

The old Dynarch DHTML date picker had a setup option 'electric' which stopped the date input field from updating until a specific date was picked.  This was not implement in the new joomla.calendar.  This PR reinstates this functionality and adds support for it into JFormFieldCalendar

The reason its important is that calendar/event components will have javascript events triggered off changes in date fields - if these are changed spuriously as the date picker is navigated it can lead to unforseen/unplanned for outcomes such as date validation errors or incorrect data entry.

### Testing Instructions

1. Edit/create and article, go to the publishing tab and click the date picker button for the start date.  Navigate the months and years and notice that the date picker field changes value.  This is the 'electric; behaviour in action.

2. Edit the file administrator/components/com_content/models/forms/article.xml and add 

`                        electric="false"`

to the publish_up field at line c. 166 and repeat step 1.

3. Edit the file administrator/components/com_content/models/forms/article.xml and add 

`                        electric="true"`

to the publish_up field at line c. 166 and repeat step 1.

### Expected result

1. Date input field changes value as you navigate.

2. Date input field does NOT change value as you navigate.

3. Date input field changes value as you navigate.

### Actual result

1. Date input field changes value as you navigate.

2. Date input field changes value as you navigate.

3. Date input field changes value as you navigate.

In fact the joomla.calendar javascript has the 'electric' behaviour hard coded in.

### Documentation Changes Required

Not sure if 'electric' is the best parameter name/description - its just that is what the old calendar used.
